core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -62,7 +62,7 @@
  * Use {@link #builderFor(FileIO, Iterable)} to construct an index, and {@link #forDataFile(long, DataFile)} or
  * {@link #forEntry(ManifestEntry)} to get the the delete files to apply to a given data file.
  */
-class DeleteFileIndex {
+public class DeleteFileIndex {
```

Comment:
Public again because we need to access this class from the Spark package in order to do our planning

Reply:
This should be okay, as long as we aren't exposing `ManifestEntry`.